packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -161,18 +158,17 @@ export class LibraryDeploySourcePathExecutor extends LibraryCommandletExecutor<
       const parser = new LibraryDeployResultParser(result);
       const outputResult = parser.resultParser(result);
       channelService.appendLine(outputResult);
-
+      BaseDeployExecutor.errorCollection.clear();
```

Comment:
it looks like the diagnostics get cleared at the beginning of `handleDeployRetrieveLibraryDiagnostics()` anyway, so do we need to clear them here as well?

Reply:
Yes because `handleDeployRetrieveLibraryDiagnostics()` is only called when the status is not Succeeded or Completed